bindings/xen/platform.c

```diff
@@ -295,6 +341,30 @@ void platform_init(const void *arg __attribute__((unused)))
             }
         }
     }
+}
+
+void platform_init(const void *arg)
+{
+    bool multiboot = *((uint32_t*)arg) != XEN_HVM_START_MAGIC_VALUE;
+
+    /*
+     * Parse multiboot before setting page tables, as its location may not be
+     * mapped later.
+     */
+    if (multiboot)
+        parse_multiboot(arg);
+
+    pagetable_init();
+    hypercall_init();
+    console_init();
+
+    /*
+     * Parse Xen HVM/PVH start_info structure after setting hypercall page - it
```

Comment:
"If booted directly, parse the Xen hvm_start_info structure after setting up hypercalls"